gsonpath-compiler/src/main/java/gsonpath/model/GsonObjectFactory.kt

```diff
@@ -1,26 +1,22 @@
 package gsonpath.model
 
-import com.google.gson.FieldNamingPolicy
 import com.google.gson.annotations.SerializedName
 import com.squareup.javapoet.TypeName
 import gsonpath.GsonFieldValidationType
-import gsonpath.PathSubstitution
 import gsonpath.ProcessingException
-import gsonpath.generator.standard.SubTypeMetadata
-import gsonpath.generator.standard.SubTypeMetadataFactory
+import gsonpath.generator.standard.subtype.SubTypeMetadata
+import gsonpath.generator.standard.subtype.SubTypeMetadataFactory
 import java.util.regex.Pattern
 import javax.lang.model.element.Element
 
 class GsonObjectFactory(private val subTypeMetadataFactory: SubTypeMetadataFactory) {
 
     @Throws(ProcessingException::class)
-    fun addGsonType(gsonPathObject: GsonObject,
-                    fieldInfo: FieldInfo,
-                    fieldInfoIndex: Int,
-                    flattenDelimiter: Char,
-                    gsonFieldNamingPolicy: FieldNamingPolicy,
-                    gsonFieldValidationType: GsonFieldValidationType,
-                    pathSubstitutions: Array<PathSubstitution>) {
+    fun addGsonType(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=kotlin%3AS3776)